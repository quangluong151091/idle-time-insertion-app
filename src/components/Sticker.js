import React from 'react';

class Sticker extends React.Component {
    onDragStart = (ev, id) => {
        console.log("dragstart:", id);
        var item = {
            content: id
        };
        ev.dataTransfer.setData("text", JSON.stringify(item));
    }
    render() {
        const imgPath = '/pics/stickers/' + this.props.img + '.png'
        return (
            <td style={{ width: '50%' }}>
                <img className="item" src={imgPath} alt={this.props.name} draggable
                    onDragStart={(e) => this.onDragStart(e, this.props.img)} /><br />
                <span>{this.props.name.toUpperCase()}</span>
            </td>
        )
    }
}

export default Sticker;