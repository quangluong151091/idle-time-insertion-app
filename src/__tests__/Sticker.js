import React from 'react';
import { shallow } from 'enzyme';

import Sticker from '../components/Sticker';

describe('Sticker', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Sticker id="1" img="ic-icon" name="Tool" />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be draggable', () => {
    wrapper.find(".item").simulate("dragstart", { dataTransfer: { setData: jest.fn() } });
  })
});