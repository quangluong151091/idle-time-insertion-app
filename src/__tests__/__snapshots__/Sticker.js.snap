// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticker should render correctly 1`] = `
<td
  style={
    Object {
      "width": "50%",
    }
  }
>
  <img
    alt="Tool"
    className="item"
    draggable={true}
    onDragStart={[Function]}
    src="/pics/stickers/ic-icon.png"
  />
  <br />
  <span>
    TOOL
  </span>
</td>
`;
